### PR TITLE
Refactor common executor

### PR DESCRIPTION
**Pull request by x3medima17**
Signed-off-by: Dumitru <dimasavva17@gmail.com>

### Description of the Change
Made `getAccountPermissions` and `checkAccountRolePermission` functional.

### Benefits
No side effects
No object copying, thus faster runtime,

### Possible Drawbacks 
None

### CODE REVIEW

File: irohad/execution/impl/common_executor.cpp [hunk 1]
```diff
@@ -27,37 +27,34 @@ namespace iroha {
...
+    return queries.getAccountRoles(account_id) | [&queries](const auto &roles) {
+      return std::accumulate(roles.begin(),
+                             roles.end(),
+                             shared_model::interface::RolePermissionSet{},
+                             [&queries](auto &&permissions, const auto &role) {
+                               queries.getRolePermissions(role) |
+                                   [&](const auto &perms) {
+                                     permissions |= perms;
+                                   };
+                               return permissions;
```
Comment (kamilsa) [inline irohad/execution/impl/common_executor.cpp]: won't it make a copy?

Comment (l4l) [reply irohad/execution/impl/common_executor.cpp]: yeah, would be nice to add std::forward

File: irohad/execution/impl/common_executor.cpp [hunk 2]
```diff
@@ -27,37 +27,34 @@ namespace iroha {
...
-      if (not rolePerms)
-        continue;
-      if (rolePerms->test(permission)) {
-        return true;
-      }
-    }
-    return false;
+    return queries.getAccountRoles(account_id) |
+        [&permission, &queries](const auto &accountRoles) {
+          return std::any_of(accountRoles.begin(),
```
Comment (l4l) [inline irohad/execution/impl/common_executor.cpp]: :+1:

File: irohad/execution/impl/common_executor.cpp (hunk 1 above)
Comment (kamilsa) [reply irohad/execution/impl/common_executor.cpp]: Then decltype auto should be added as well

File: irohad/execution/impl/common_executor.cpp [hunk 3]
```diff
@@ -28,16 +28,16 @@ namespace iroha {
...
-                                   };
-                               return permissions;
-                             });
+      return std::accumulate(
+          roles.begin(),
+          roles.end(),
+          shared_model::interface::RolePermissionSet{},
+          [&queries](auto &&permissions, const auto &role) {
+            queries.getRolePermissions(role) |
+                [&](const auto &perms) { permissions |= perms; };
```
Comment (l4l) [inline irohad/execution/impl/common_executor.cpp]: formatting :(

**Pull request closed without merging**